apps/web-giddh/src/app/shared/advance-receipt-adjustment/advance-receipt-adjustment.component.ts

```diff
@@ -582,12 +652,45 @@ export class AdvanceReceiptAdjustmentComponent implements OnInit {
     public checkValidations(): void {
         if (this.adjustVoucherForm && this.adjustVoucherForm.adjustments && this.adjustVoucherForm.adjustments.length > 0) {
             this.adjustVoucherForm.adjustments.forEach((item, key) => {
-                if ((!item.voucherNumber && item.dueAmount.amountForAccount) || (item.voucherNumber && !item.dueAmount.amountForAccount) || (!item.voucherNumber && !item.dueAmount.amountForAccount && key > 0)) {
+                if ((!item.voucherNumber && item.balanceDue.amountForAccount) || (item.voucherNumber && !item.balanceDue.amountForAccount) || (!item.voucherNumber && !item.balanceDue.amountForAccount && key > 0)) {
                     this.isInvalidForm = true;
                 } else {
                     this.isInvalidForm = false;
                 }
             });
         }
     }
+
+    /**
+     * Returns true if the voucher adjustment EDIT operation is supported
+     *
+     * @readonly
+     * @type {boolean} True if the voucher adjustment EDIT operation is supported
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    public shouldDisableEdit(item: Adjustment): boolean {
+        return this.isVoucherModule && item.voucherType && !(item.voucherType === 'receipt' && item.subVoucher === SubVoucher.AdvanceReceipt);
+    }
+
+    /**
+     * Used to format data as the adjustments array have two keys as
+     * 'balanceDue' and 'adjustmentAmount'. Former is received when any voucher
+     * is adjusted during creating entry and represents the BALANCE amount that can
+     * get adjusted and latter is received when any voucher
+     * is already adjusted and represents the ADJUSTED amount
+     *
+     * @private
+     * @memberof AdvanceReceiptAdjustmentComponent
+     */
+    private formatAdjustmentData(adjustmentData: Array<Adjustment>): void {
+        if (adjustmentData && adjustmentData.length) {
+            adjustmentData.forEach(adjustment => {
+                if (adjustment.adjustmentAmount && adjustment.adjustmentAmount.amountForAccount) {
```

Comment:
Please check this condition if  "adjustment.adjustmentAmount.amountForAccount" value will be zero then we not allowing to update balance due